samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -130,11 +183,19 @@
    * Note: The message will be written but not flushed to the underlying output system before its propagated to the
    * chained operators. Messages retain the original partitioning scheme when propogated to next operator.
    *
+   * @param desc description of this send to operation
    * @param outputStream the output stream to send messages to
    * @return this {@link MessageStream}
    */
-  MessageStream<M> sendTo(OutputStream<M> outputStream);
+  MessageStream<M> sendTo(String desc, OutputStream<M> outputStream);
 
+  /**
+   * Equivalent to {@code window("", window, id)}
+   * @see #window(String, Window, String)
+   */
+  default <K, WV> MessageStream<WindowPane<K, WV>> window(Window<M, K, WV> window, String id) {
```

Comment:
IIUC, windows, joins, partitionBy etc already have the "id" param which has the same function. Do we need the new param as well?

Would also prefer calling the param "id" instead of "description" to clarify that it should be concise, and keeping it at the last param for consistency.

Reply:
Good question! The reason why I added to window, join and partition by is because, one major use case of high level API in Samza is to support Beam API, where the existing `id` are auto-generated for users. i.e. it does not help a lot when users tries to associate their transforms with operator id, therefore, a customized name would greatly help.

Reply:
Couldn't we pass the custom name that the user provided to the Beam API  (if one exists) as the id here _instead_ of the Beam auto-generated id?

Reply:
We could but that will be a backward incompatible change that breaks all stateful jobs.

Reply:
"Backward incompatible for stateful" because the changelog name maps to the operator name?

Reply:
@kw2542 Do you think we can safely manage this backwards incompatible change (e.g. choosing safer defaults, making it opt-in, providing migration tooling etc.)? Would be better to manage a one time change than add a permanent new API to work around it.

Reply:
After some more thoughts, I agree that providing a name param for these stateful operators does not help with backward compatibility. I will exclude them from the API change.